service/history/api/respondactivitytaskfailed/api.go

```diff
@@ -152,3 +154,13 @@ func Invoke(
 	}
 	return &historyservice.RespondActivityTaskFailedResponse{}, err
 }
+
+func nextRetryDelayFrom(request *workflowservice.RespondActivityTaskFailedRequest) *time.Duration {
+	p := request.GetActivityRequests().GetNextRetryDelay()
+	var delay *time.Duration
+	if p != nil {
+		d := p.AsDuration()
+		delay = &d
+	}
+	return delay
```

Comment:
The idiomatic way to do this would be to return the zero-valued `time.Duration` (0) rather than a nil pointer to indicate the lack of a delay unless the difference between an unspecified duration (nil in your code) and a 0 duration is meaningful